code/model/editableformfields/EditableFormField.php

```diff
@@ -365,6 +365,22 @@ public function canEdit($member = null) {
         $parent = $this->Parent();
 		if($parent && $parent->exists()) {
 			return $parent->canEdit($member) && !$this->isReadonly();
+		} else if ($this->ID == 0) {
+			// This is for GridFieldOrderableRows support as it checks edit permissions on 
+			// singleton of the class. Allows editing of User Defined Form pages by 
+			// 'Content Authors' and those with permission to edit the UDF page. (ie. CanEditType/EditorGroups)
+			// This is to restore User Forms 2.x backwards compatibility.
+			$controller = Controller::curr();
+			if ($controller && $controller instanceof CMSPageEditController)
```

Comment:
Controller::curr() currently fails if there is no controller. Please use Controller::has_curr() as a substitute check for this first condition. :)